src/WorkflowManager/WorkflowManager/Controllers/WorkflowsController.cs

```diff
@@ -261,6 +261,14 @@ public async Task<IActionResult> UpdateAsync([FromBody] WorkflowUpdateRequest re
 
             try
             {
+                if (workflow.Name != originalName)
+                {
+                    foreach (var task in workflow.Tasks)
+                    {
+                        task.Args["workflow_name"] = workflow.Name;
+                    }
+                }
+
```

Comment:
Could this not be done in the service for each task like you've done it with the create one?

Reply:
I've updated the code moved logic into service.